src/Swaps/SwapVerifier.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import {IEVault, IERC20} from "evk/EVault/IEVault.sol";
+
+/// @title SwapVerifier
+/// @custom:security-contact security@euler.xyz
+/// @author Euler Labs (https://www.eulerlabs.com/)
+/// @notice Simple contract used to verify post swap conditions
+/// @dev This contract is the only trusted code in the EVK swap periphery
+contract SwapVerifier {
+    error SwapVerifier_skimMin();
+    error SwapVerifier_debtMax();
+
+    /// @notice Verify amount of assets available for `skim` in an EVault is greater than a given limit
+    /// @param vault The EVault to query
+    /// @param amountMin Minimum amount of assets that should be available for skim
+    /// @dev Swapper contract will send bought assets to the vault in certain situations.
+    /// @dev Calling the function is then equivalent to a slippage check.
+    function verifySkimMin(address vault, uint256 amountMin) external view {
```

Comment:
Why do we care so much about this contract being separate? If those functions were in the Swapper, it would be much cheaper to call them using the Swapper multicall rather than going through the EVC auth. I know that we're saying that the Swapper is not a part of a trusted code base etc., but why can't we consider it a part of a trusted code base if we'll audit it anyways?

You could even simplify this and rather than verifying and then calling skim (two EVC batch items), you could just call this function on the Swapper:

```solidity
function skim(address token, address vault, uint256 amountMin, address account) public external Lock {
   uint256 cash = IEVault(vault).cash();
   uint256 balance = IERC20(token).balanceOf(vault);

   if (balance <= cash || balance - cash < amountMin) revert();

   IEVault(vault).skim(type(uint256).max, account);
}
```